plugins/node/opentelemetry-instrumentation-dns/test/integrations/dns-lookup.test.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/tracing';
+import * as assert from 'assert';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import { DnsInstrumentation } from '../../src/dns';
+import * as dns from 'dns';
+import * as utils from '../utils/utils';
+import { assertSpan } from '../utils/assertSpan';
+import { StatusCode } from '@opentelemetry/api';
+
+const memoryExporter = new InMemorySpanExporter();
+const provider = new NodeTracerProvider();
+provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+
+describe('dns.lookup()', () => {
+  let instrumentation: DnsInstrumentation;
+
+  before(function (done) {
+    // mandatory
+    if (process.env.CI) {
+      instrumentation = new DnsInstrumentation();
+      instrumentation.setTracerProvider(provider);
+      require('dns');
```

Comment:
doesnt the `dns` package already required l.24 ?

Reply:
Seems it can be removed here but not at other places in tests otherwise instrumentation doesn't happen.
Most likely related to how require-in-the-middle and the instrumentation base classes interact with require cache.